utils/prisma/prismaHelpers.ts

```diff
@@ -0,0 +1,47 @@
+import { Prisma } from '@prisma/client';
+import { NextApiResponse } from 'next';
+import { StatusMessageType } from '../../types/apiResponse';
+import { createJsonResponse, HTTP_STATUS_BAD_REQUEST, HTTP_STATUS_INTERNAL_SERVER_ERROR } from '../http/httpHelper';
+
+/**
+ * Utility function that takes care of Prisma runtime errors during CREATE.
+ *
+ * @param res The NextApiResponse object.
+ * @param creator The Prisma Creator function.
+ */
+export function createIfPossible(res: NextApiResponse, creator: () => Promise<void>) {
+  creator().catch((error) => {
+    console.error(error);
+
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      res.status(HTTP_STATUS_BAD_REQUEST).json(
+        createJsonResponse({}, [
+          {
+            type: StatusMessageType.Error,
+            message: 'A database error has occured due to the request. Please try again.',
+          },
+        ]),
+      );
+      return;
+    }
+
+    if (error instanceof Prisma.PrismaClientValidationError) {
+      res
+        .status(HTTP_STATUS_BAD_REQUEST)
+        .json(
+          createJsonResponse({}, [
+            { type: StatusMessageType.Error, message: 'Missing fields in request. Please try again.' },
```

Comment:
This may not only be due to missing fields right? https://www.prisma.io/docs/reference/api-reference/error-reference#prismaclientvalidationerror

Reply:
Could be due to bad type in field too, nice catch.